app/score_calculator/yaku_check/hand_yaku_checker.py

```diff
@@ -0,0 +1,92 @@
+from collections import defaultdict
+from collections.abc import Callable
+from functools import cached_property
+
+from app.score_calculator.block.block import Block
+from app.score_calculator.enums.enums import BlockType, Tile, Yaku
+from app.score_calculator.yaku_check.yaku_checker import YakuChecker
+
+
+# yaku checker for hand property
+class HandYakuChecker(YakuChecker):
+    def __init__(self, blocks: list[Block]):
+        super().__init__()
+        self.blocks: list[Block] = blocks
+        self._yakus: list[Yaku]
+        self.conditions: dict[str, list[tuple[bool, Yaku]]] = {
+            "num_compair": [
+                (self.is_upper_tiles, Yaku.UpperTiles),
+                (self.is_middle_tiles, Yaku.MiddleTiles),
+                (self.is_lower_tiles, Yaku.LowerTiles),
+                (self.is_upper_four, Yaku.UpperFour),
+                (self.is_lower_four, Yaku.LowerFour),
+            ],
+        }
+        self.set_yakus()
+
+    @property
+    def yakus(self) -> list[Yaku]:
+        return self._yakus
+
+    def set_yakus(self) -> None:
+        self._yakus = self.blocks_checker()
+
+    def blocks_checker(self) -> list[Yaku]:
+        return list(
+            filter(
+                lambda x: x != Yaku.ERROR,
+                (
+                    next((yaku for checker, yaku in condition if checker), Yaku.ERROR)
+                    for condition in self.conditions.values()
+                ),
+            ),

```

Comment:
list filter의 경우 일반적으로 list comprehension을 쓰는 게 일반적이긴합니다.
```suggestion
            blahblah = blahblah
            return [yaku for yaku in blahblah if yaku != Yaku.ERROR]
```